### PR TITLE
Docs+

### PR DESCRIPTION
**Pull request by BenBaryoPX**
+ Improve documentation
+ typeMap now returns an empty array if type wasn't found in tree
+ Add tests for edge cases

### CODE REVIEW

File: src/arborist.js [hunk 1]
```diff
@@ -67,76 +70,103 @@ const Arborist = class {
...
-									++changesCounter;
 								}
 							}
 						} catch (e) {
 							this.logger.debug(`[-] Unable to delete node: ${e}`);
 						}
 					}
-					for (let i = 0; i < this.replacements.length; i++) {
-						const [targetNode, replacementNode] = this.replacements[i];
+					for (const [targetNode, replacementNode] of this.replacements) {
```
[the remark this answers is not in the document]
Comment (BenBaryoPX) [reply src/arborist.js:150]: Good suggestion, but not important right now
